pkg/controllers/smart.go

```diff
@@ -2,51 +2,57 @@ package controllers
 
 import (
 	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
 	"github.com/dghubble/go-twitter/twitter"
 	"github.com/gorilla/mux"
 	"github.com/jinzhu/gorm"
-	log "github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus"
+	"github.com/smartatransit/scrapedumper/pkg/postgres"
 	"github.com/smartatransit/third_rail/pkg/clients"
 	"github.com/smartatransit/third_rail/pkg/models"
-	"net/http"
-	"strconv"
 )
 
+type MartaMirror interface {
+	GetLatestEstimates(stationID uint) (res []postgres.LastestEstimate, err error)
+}
+
 type SmartController struct {
 	MartaClient   clients.MartaClient
 	TwitterClient clients.TwitterClient
+
+	SDClient MartaMirror
 }
 
 func (controller SmartController) GetStationDetails(db *gorm.DB, w http.ResponseWriter, req *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 
 	v := mux.Vars(req)
-
 	stationId, idErr := strconv.Atoi(v["id"])
 	if idErr != nil {
 		w.WriteHeader(http.StatusUnprocessableEntity)
 		return
 	}
 
 	var station models.Station
-	db.Preload("Detail").Preload("Lines").Preload("Aliases").First(&station, stationId)
-
-	scheduleEvents, realTimeDetails, seErr := models.GetScheduleEventsByStationRealTime(stationId, db)
+	if err := db.First(&station, stationId).Error; err != nil {
+		logrus.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
 
-	if seErr != nil {
-		log.Error(seErr)
+	sdEstimates, err := controller.SDClient.GetLatestEstimates(uint(stationId))
```

Comment:
This is calling the code [here](https://github.com/smartatransit/scrapedumper/blob/master/pkg/postgres/repository.go#L455). There's a lot that can be done to make this more efficient but it gets the job done!